src/files/compose/validateDappnodeCompose.ts

```diff
@@ -0,0 +1,273 @@
+  if (!isCore && cpServicePrivileged && cpServicePrivileged === true)
+    aggregatedError.errors.push(
+      Error(
+        `Compose service ${cpServiceName} has privileged as true but is not a core package`
+      )
+    );
+
+  // Check Only core packages can use network_mode: host
+  const cpServiceNetworkMode = compose.services[cpServiceName].network_mode;
+  if (!isCore && cpServiceNetworkMode && cpServiceNetworkMode === "host")
+    aggregatedError.errors.push(
+      Error(
+        `Compose service ${cpServiceName} has network_mode: host but is not a core package`
+      )
+    );
+}
+
+/**
+ * Ensure the compose services networks are whitelisted
+ */
+function validateComposeServiceNetworks(
+  compose: Compose,
+  isCore: boolean,
+  cpServiceName: string
+): void {
+  const cpService = compose.services[cpServiceName];
+  const cpServiceNetworks = cpService.networks;
+  if (!cpServiceNetworks) return;
+
+  for (const cpServiceNetwork of cpServiceNetworks) {
+    if (!cpServiceNetwork) continue;
+
+    if (typeof cpServiceNetwork === "string") {
+      // Check docker network is whitelisted when defined in array format
+      if (!params.DOCKER_WHITELIST_NETWORKS.includes(cpServiceNetwork))
+        aggregatedError.errors.push(
+          Error(
+            `Compose service ${cpServiceName} has a non-whitelisted docker network. Only docker networks ${params.DOCKER_WHITELIST_NETWORKS.join(
+              ","
+            )} are allowed`
+          )
+        );
+    } else {
+      const cpServiceNetworkObjectName = Object.keys(cpServiceNetwork)[0];
+      if (!cpServiceNetworkObjectName) continue;
+
+      // Check docker network is whitelisted when defined in object format
+      if (
+        !params.DOCKER_WHITELIST_NETWORKS.includes(cpServiceNetworkObjectName)
+      )
+        aggregatedError.errors.push(
+          Error(
+            `Compose service ${cpServiceName} has a non-whitelisted docker network: ${cpServiceNetworkObjectName}. Only docker networks ${params.DOCKER_WHITELIST_NETWORKS.join(
+              ","
+            )} are allowed`
+          )
+        );
+
+      // Check core aliases are not used by non core packages
+      const cpServiceNetworkAliases =
+        cpServiceNetwork[cpServiceNetworkObjectName].aliases;
+      if (
+        !isCore &&
+        cpServiceNetworkAliases &&
+        params.DOCKER_CORE_ALIASES.some(coreAlias =>
+          cpServiceNetworkAliases.includes(coreAlias)
+        )
+      ) {
+        aggregatedError.errors.push(
+          Error(
+            `Compose service ${cpServiceName} has a reserved docker alias. Aliases ${params.DOCKER_CORE_ALIASES.join(
+              ","
+            )} are reserved to core packages`
+          )
+        );
+      }
+    }
+  }
+}
+
+/**
+ * Ensure only core packages can use bind-mounted volumes
+ */
+function validateComposeAndComposeServiceVolumes(
+  compose: Compose,
+  isCore: boolean,
+  cpServiceName: string,
+  cpServiceVolume: string
```

Comment:
```suggestion
  volumeName: string
```

Reply:
it is not the name of the volume but the whole volume definition `some_volume:/some/path`
It also belongs to the service. `serviceVolume` makes more sense for me

Reply:
Fix by passing as an argument the `volumeName` instead of the `serviceVolume`